sdk/core/azure-core/inc/azure/core/internal/diagnostics/log.hpp

```diff
@@ -11,24 +11,30 @@
 
 namespace Azure { namespace Core { namespace Diagnostics { namespace _internal {
   class Log final {
-    using LogLevelInt = std::underlying_type<Logger::Level>::type;
+    static_assert(
+        std::is_same<int, std::underlying_type<Logger::Level>::type>::value == true,
+        "Logger::Level values must be representable as lock-free");
 
+    // Ensure that the LogLevel is of size that is compatible with cmpxchg16b
+    // If this is true then a reasonable implementation of atomic will be lock_free
     static_assert(
-        std::is_same<int, LogLevelInt>::value == true && ATOMIC_INT_LOCK_FREE == 2,
+        sizeof(Logger::Level) <= 8 && (sizeof(Logger::Level) & (sizeof(Logger::Level) - 1)) == 0,
```

Comment:
Remove this assertion - it is hard to read, it is an overkill, and it assumes way too much, about `cmpxchg16b`, but what about ARMs? They don't have this instruction. No, if we decide to assume that an enum with underlying type of int is stored exactly as atomic int, plus we check that int is atomic, then the other `static_assert`s here are sufficient, this one is not necessary. I'd say this assert is harder to read than a static_cast.